cpptod.dd

```diff
@@ -829,6 +830,84 @@ void test()
 }
 ------
 
+<hr>$(COMMENT  -------------------------------------------- )
+
+$(H3 <a name="interfaces">Interfaces</a>)
+
+Interfaces are used to describe the behavior or capabilities of a class without
+commiting to a particular implementation of the class.
+
+$(H4 The C++ Way)
+
+  In C++ an interface is implemented with an `Abstract Base Class`. The ABC defines
+  the class capabilities as pure virtual functions. A class which implements
+  the interface inherits from the ABC and implements the pure virtual functions.
+
+$(CPPCODE
+class Interface {
+  public:
+    virtual int function() = 0;
+};
+
+class FirstVariantOfImplementation : public Interface {
+  public:
+    int function() {
+      ...
+    }
+}
+
+class SecondVariantOfImplementation : public Interface {
+  public:
+    int function() {
+      ...
+    }
+}
+)
+
+The interface is used as follows:
+
+$(CPPCODE
+FirstVariantOfImplementation FirstVariant;
+SecondVariantOfImplementation SecondVariant;
+
+FirstVariant.function();
+SecondVariant.function();
+)
+
+$(H4 The D Way)
+
+  In D interfaces are implemented using the `interface keyword`.
+  The implementation class inherits from the interface and overrides the
+  interface functions.
```

Comment:
Currently `override` is necessary only when overriding an inherited function from another class (doesn't matter if the function is pure virtual  or not). When you're implementing an interface, `override` is not necessary.

@andralex should `override` be necessary when implementing interface functions? If not, should the compiler warn about that? IMO it is inconsistent to require `override` when implementing pure virtual functions declared by abstract base classes, but not by interfaces - it should be either both or none.